### ISSUE TITLE
UnsupportedOperationException GLES20Canvas.clipPath in 4.0

### ISSUE THREAD

**Issue by 0kai** [whilu/AndroidTagView#50]
```
1 android.view.GLES20Canvas.clipPath(GLES20Canvas.java:424)
2 co.lujun.androidtagview.TagView.void drawRipple(android.graphics.Canvas)(SourceFile:329)
3 co.lujun.androidtagview.TagView.void onDraw(android.graphics.Canvas)(SourceFile:189)
4 android.view.View.draw(View.java:10988)
5 android.view.View.getDisplayList(View.java:10427)
6 android.view.ViewGroup.drawChild(ViewGroup.java:2850)
7 android.view.ViewGroup.dispatchDraw(ViewGroup.java:2489)
8 android.view.View.draw(View.java:10991)
9 android.view.View.getDisplayList(View.java:10427)
10 android.view.ViewGroup.drawChild(ViewGroup.java:2850)
11 android.view.ViewGroup.dispatchDraw(ViewGroup.java:2489)
12 android.view.View.getDisplayList(View.java:10425)
13 android.view.ViewGroup.drawChild(ViewGroup.java:2850)
14 android.support.v7.widget.RecyclerView.drawChild(SourceFile:3838)
15 android.view.ViewGroup.dispatchDraw(ViewGroup.java:2489)
```

**Comment by whilu** [comment]
@0kai 
There are some problem about hardware acceleration, you can disable hardware acceleration in your 'Activity' which used AndroidTagView.  
I will improve this in the next version.
Thanks! 😄

**Comment by whilu** [comment]
This problem has been improved in v1.1.1, so I close the issue. Thanks!

**Comment by 0kai** [comment]
catch a same issue after update to v1.1.1
![image](https://cloud.githubusercontent.com/assets/1498653/26041056/cb0531a2-395c-11e7-98be-5254642646e0.png)

Does the problem only been improved, may reproduce ?

**Comment by whilu** [comment]
The `setLayerType(View.LAYER_TYPE_SOFTWARE, null)` method  looks useless in U8825D?

Look this: [LAYER_TYPE_SOFTWARE still view drawn with hardware acceleration](https://issuetracker.google.com/issues/36933727)

I don't have this phone and tested on the emulator last time, It's OK on emulator because there is no hardware acceleration.

Finally, I add protection for `UnsupportedOperationException`, but there is no ripple effect anymore(Of course, there shouldn't be have).

I released v1.1.2, and I hope you can help with the test if you have that phone(or phone with Android API level 11-17). That would be great!

Thanks!

**Comment by 0kai** [comment]
OK, but I do not have the U8825D phone, got it from crash report from market.
And I will update this library to v1.1.2, if any new message will let you know.

Thanks!
